dataland-frontend/src/components/resources/frameworkDataSearch/lksg/LksgPanel.vue

```diff
@@ -145,10 +157,13 @@ export default defineComponent({
             dataId: dataIdOfLksgDataset,
             reportingPeriod: reportingPeriodOfLksgDataset,
           });
-          for (const areaObject of Object.values(oneLksgDataset.data)) {
+          for (const [areaKey, areaObject] of Object.entries(oneLksgDataset.data)) {
             for (const [subAreaKey, subAreaObject] of Object.entries(areaObject as object) as [string, object][]) {
               for (const [kpiKey, kpiValue] of Object.entries(subAreaObject) as [string, object][]) {
-                this.createKpiDataObjects(kpiKey, kpiValue, subAreaKey, dataIdOfLksgDataset);
+                const subcategory = lksgDataModel
+                  .filter((area) => area.name === areaKey)[0]
```

Comment:
I discovered various 
`xxx.filter(...)[0]` patterns in the code. They should be replaced with 
`xxx.find(...)` - as find already returns the first. If things go well, it's even faster as it's lazy.

Reply:
See the above case. "filter" provides easier undefined handling. But performance is of course a good argument for "find".
Can we discuss this?

Reply:
A way to implement the "find" logic would be:
`assertDefined(xxx.find(...))`

where `assertDefined` is our custom utils function with the sole purpose to trick the linter that usually forbids "non-null assertions".

Reply:
replaced with assertDefined(xx.find(...))

Reply:
There's still this pattern in Tooltips.ts and SearchCompaniesForFrameworkData.ts and some other places. Could you recheck?

Reply:
fixed them

Reply:
ok